hedera-node/src/main/java/com/hedera/services/state/expiry/removal/AccountGC.java

```diff
@@ -96,16 +106,46 @@ public TreasuryReturns expireBestEffort(final EntityNum expiredAccountNum, final
 					tokenTypes,
 					returnTransfers,
 					tokenRels.get());
+			// set num rels to 0 as we remove all associations
+			account.setNumAssociations(0);
 		}
 
-		backingAccounts.remove(expiredAccountNum.toGrpcAccountId());
-		sigImpactHistorian.markEntityChanged(expiredAccountNum.longValue());
-		if (aliasManager.forgetAlias(account.getAlias())) {
-			aliasManager.forgetEvmAddress(account.getAlias());
-			sigImpactHistorian.markAliasChanged(account.getAlias());
+		if (!done) {
+			final var nftsOwned = account.getNftsOwned();
+			done = returnNfts(
+					nftsOwned,
+					account.getHeadNftId(),
+					account.getHeadNftSerialNum(),
+					uniqueTokens.get());
+			final var remainingNfts = done ? 0 : nftsOwned - dynamicProperties.getMaxReturnedNftsPerTouch();
+			account.setNftsOwned(remainingNfts);
 		}
 
-		return new TreasuryReturns(tokenTypes, returnTransfers, true);
+		if (done) {
+			backingAccounts.remove(expiredAccountNum.toGrpcAccountId());
+			sigImpactHistorian.markEntityChanged(expiredAccountNum.longValue());
+			if (aliasManager.forgetAlias(account.getAlias())) {
+				aliasManager.forgetEvmAddress(account.getAlias());
+				sigImpactHistorian.markAliasChanged(account.getAlias());
+			}
+		}
+
+		return new TreasuryReturns(tokenTypes, returnTransfers, done);
+	}
+
+	private boolean returnNfts(
+			final long nftsOwned,
+			final long headNftNum,
+			final long headSerialNum,
+			final MerkleMap<EntityNumPair, MerkleUniqueToken> currUniqueTokens
+	) {
+		final var uniqueTokensRemoval = new UniqueTokensListRemoval(currUniqueTokens);
+		var nftKey = EntityNumPair.fromLongs(headNftNum, headSerialNum);
+		var i = Math.min(nftsOwned, dynamicProperties.getMaxReturnedNftsPerTouch());
+		while (nftKey != null && i-- > 0) {
+			nftKey = treasuryReturnHelper.updateNftReturns(nftKey, uniqueTokensRemoval);
+		}
+		return nftsOwned < dynamicProperties.getMaxReturnedNftsPerTouch();
```

Comment:
this check can be done after/before this function , since the values of it are not changing ?